.github/workflows/build_test.yml

```diff
@@ -3,45 +3,30 @@ name: ROS2 CI
 on: [pull_request, push]
 
 jobs:
-  test_environment:
-    runs-on: [ubuntu-latest]
+  build-and-test:
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         ros_distribution:
-          - foxy
           - galactic
-          - rolling
+          - humble
+
         include:
-          # Foxy Fitzroy (May 2020 - May 2023)
-          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
-            ros_distribution: foxy
-            ros_version: 2
-          # Galactic Geochelone (May 2021 - November 2022)
-          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-galactic-ros-base-latest
+          - docker_image: ubuntu:focal
             ros_distribution: galactic
-            ros_version: 2
-          # Rolling Ridley  (June 2020 - Present)
-          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-rolling-ros-base-latest
-            ros_distribution: rolling
-            ros_version: 2
-        linter:
-          - copyright
-          - xmllint
-          - lint_cmake
+          - docker_image: ubuntu:jammy
+            ros_distribution: humble
+
     container:
       image: ${{ matrix.docker_image }}
+
     steps:
-      - name: setup directories
-        run: mkdir -p ros_ws/src
-      - name: checkout
-        uses: actions/checkout@v2
+      - name: setup ROS environment
+        uses: ros-tooling/setup-ros@v0.4
         with:
-          path: ros_ws/src
-      - name: build and test
-        continue-on-error: true
-        uses: ros-tooling/action-ros-ci@0.2.2
+          required-ros-distributions: ${{ matrix.ros_distribution }}
+      - name: build and test ROS 2
+        uses: ros-tooling/action-ros-ci@v0.2
         with:
-          package-name: udp_driver serial_driver
-          target-ros2-distro: ${{ matrix.ros_distribution }}
-          vcs-repo-file-url: ""
+          target-ros2-distro: ${{ matrix.ros_distribution }}
```

Comment:
Could you open a separate PR for these changes? They don't seem related to supporting TCP and HTTP

Reply:
Addressed [here](https://github.com/ros-drivers/transport_drivers/pull/84/commits/32943067c12da460a98d4792e5cad3914ae8d2b7)

Reply:
Thanks. Could you move these changes to a separate PR? I think they're useful, just outside of the scope of this PR